### PR TITLE
Shop Hooking!

### PR DESCRIPTION
**Pull request by Saturn745**
This PR introduces hooking into different shop plugins for selling the loot in spawners

- [x] Essentials
- [x] EconomyShopGUI (Tested with the Premium version. Free version has not been tested but should work fine) 
- [x] PeddlersPocket (Waiting for a API method to get the sell amount for an item)

### CODE REVIEW

File: src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java [hunk 1]
```diff
@@ -89,4 +110,38 @@ public void onDisable() {
...
+        this.vaultEconomy = rsp.getProvider();
+        return this.vaultEconomy != null;
+    }
+    private boolean loadShops() {
+        // This could probably be improved
+        shopPlugins.put("Essentials", EssentialsShop.class);
+        shopPlugins.put("EconomyShopGUI", EconomyShopGUI.class);
+        shopPlugins.put("EconomyShopGUI-Premium", EconomyShopGUI.class);
+        shopPlugins.put("PeddlersPocket", EconomyShopGUI.class);
+        Class<? extends ShopHook> pluginClass = shopPlugins.get(this.getConfig().getString("shop-plugin"));
```
Comment (xhyrom) [inline src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java]: We can use `switch` statement instead of using array and then putting values into it.

**Review by xhyrom — CHANGES_REQUESTED** [review]

File: src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java [hunk 2]
```diff
@@ -35,7 +47,16 @@ public void onLoad() {
     public void onEnable() {
         CommandAPI.onEnable();
         saveDefaultConfig();
-
+        if (!setupVault()) {
```
Comment (xhyrom) [inline src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java]: We can move this things into hooks/Hooks instead of flooding main class with this stuff.

**Review by xhyrom — CHANGES_REQUESTED** [review]

**Comment by xhyrom** [comment]
You can update Peddler's Pocket to version 1.1.0 for new api methods!

```diff
- compileOnly("me.xhyrom.peddlerspocket:PeddlersPocket:1.0.1")
+ compileOnly("dev.xhyrom.peddlerspocket:PeddlersPocket:1.0.1") {
+    exclude(group = "net.kyori")
+ }
```

File: src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java (hunk 1 above)
Comment (Saturn745) [reply src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java]: I considered that as well but ended up going with this route. I will switch it to a switch statement

File: src/main/resources/config.yml [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# Available shop plugins: Essentials, EconomyShopGUI(-Premium), PeddlersPocket(TODO)
```
Comment (xhyrom) [inline src/main/resources/config.yml]: you can remove (TODO) here

File: src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java [hunk 4]
```diff
@@ -79,6 +76,7 @@ else if (action.containsKey("sound"))
         getServer().getPluginManager().registerEvents(new ClickListener(), this);
         getServer().getPluginManager().registerEvents(new SpawnerListener(), this);
         getServer().getPluginManager().registerEvents(new BlockListener(), this);
+        new Hooks();
```
Comment (xhyrom) [inline src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java]: just preference, dont we want to have `Hooks#init` static than creating class? That's just preference but there's no reason  for not using static functions

Comment (xhyrom) [reply src/main/java/me/xhyrom/spawnergenz/SpawnerGenz.java]: tell your opinion

**Pull request merged**